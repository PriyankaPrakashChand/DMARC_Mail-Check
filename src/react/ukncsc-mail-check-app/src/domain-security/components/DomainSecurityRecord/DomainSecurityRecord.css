.DomainSecurityRecord {
  word-break: break-word;
  word-wrap: break-word;
  font-family: monospace;
  color: #34363c;
  letter-spacing: 0;
  line-height: 24px;
}
