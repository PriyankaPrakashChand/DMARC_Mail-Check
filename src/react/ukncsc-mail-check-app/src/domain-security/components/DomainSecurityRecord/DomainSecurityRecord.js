import React from 'react';
import { Header } from 'semantic-ui-react';

import './DomainSecurityRecord.css';

const DomainSecurityRecord = ({ type, inheritedFrom, children }) => (
  <React.Fragment>
    <Header as="h3">
      Record{type === 'DMARC' && inheritedFrom
        ? ` inherited from ${inheritedFrom}`
        : ''}
    </Header>
    <p className="DomainSecurityRecord">{children}</p>
  </React.Fragment>
);

export default DomainSecurityRecord;
