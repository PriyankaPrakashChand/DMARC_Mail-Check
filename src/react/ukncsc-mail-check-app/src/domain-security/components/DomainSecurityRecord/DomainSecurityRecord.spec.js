import React from 'react';
import { render } from 'react-testing-library';
import 'dom-testing-library/extend-expect';
import DomainSecurityRecord from './DomainSecurityRecord';

describe('DomainSecurityRecord', () => {
  let container;

  describe('when the record is not inherited', () => {
    beforeEach(() => {
      ({ container } = render(
        <DomainSecurityRecord>record1</DomainSecurityRecord>
      ));
    });

    test('it should match the snapshot', () =>
      expect(container).toMatchSnapshot());
  });

  describe('when the record is inherited', () => {
    beforeEach(() => {
      ({ container } = render(
        <DomainSecurityRecord type="DMARC" inheritedFrom='foo.com'>
          record2
        </DomainSecurityRecord>
      ));
    });

    test('it should say so', () =>
      expect(container.querySelector('h3')).toHaveTextContent(
        'Record inherited from foo.com'
      ));
  });
});
