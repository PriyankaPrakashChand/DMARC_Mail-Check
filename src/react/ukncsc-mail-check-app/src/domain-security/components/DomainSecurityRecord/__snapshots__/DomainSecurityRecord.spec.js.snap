// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DomainSecurityRecord when the record is not inherited it should match the snapshot 1`] = `
<div>
  <h3
    class="ui header"
  >
    Record
  </h3>
  <p
    class="DomainSecurityRecord"
  >
    record1
  </p>
</div>
`;
